src/k8s-extension/azext_k8s_extension/partner_extensions/EntraWorkloadIAM.py

```diff
@@ -106,9 +126,14 @@ def Create(self, cmd, client, resource_group_name, cluster_name, name, cluster_t
         # Save configuration setting values to overwrite values in the Helm chart
         configuration_settings[CONFIG_SETTINGS_HELM_TRUST_DOMAIN] = trust_domain
         configuration_settings[CONFIG_SETTINGS_HELM_TENANT_ID] = tenant_id
-        configuration_settings[CONFIG_SETTINGS_HELM_JOIN_TOKEN] = join_token
 
-        logger.debug("Configuration settings value for Helm: %s" % str(configuration_settings))
+        if configuration_protected_settings is None:
+            configuration_protected_settings = dict()
+        configuration_protected_settings[CONFIG_SETTINGS_HELM_JOIN_TOKEN] = join_token
+
+        logger.debug("Configuration settings: %s" % str(configuration_settings))
+        logger.debug("Configuration protected settings value for Helm: %s" %
+                     str(configuration_protected_settings))
```

Comment:
These are protectedSettings, right?  Is it necessary to write this?  Can you just write the keys, without the values - just to be safe.

Reply:
Oops, right! Thanks for noticing. It wouldn't be a huge deal if it was printed, but yeah, if we go through this effort to have the setting as a protected setting we might as well hide it from here.